File: models/Gpt-3.5-turbo/cpp/code/problems/graph_algorithms/Flight_Routes_Check.cpp

#include <iostream>
#include <vector>

using namespace std;

// Function to perform DFS
void dfs(int node, vector<vector<int>>& adj, vector<bool>& visited) {
    visited[node] = true;
    for (int neighbor : adj[node]) {
        if (!visited[neighbor]) {
            dfs(neighbor, adj, visited);
        }
    }
}

int main() {
    int n, m;
    cin >> n >> m;

    vector<vector<int>> adj(n + 1); // adjacency list to represent the flights

    for (int i = 0; i < m; i++) {
        int a, b;
        cin >> a >> b;
        adj[a].push_back(b);
    }

    vector<bool> visited(n + 1, false); // array to keep track of visited cities
    dfs(1, adj, visited);

    // Check if all cities are visited
    for (int i = 1; i <= n; i++) {
        if (!visited[i]) {
            cout << "NO" << endl;
            cout << "1 " << i << endl;
            return 0;
        }
    }

    cout << "YES" << endl;

    return 0;
}

File: models/Gpt-3.5-turbo/py/code/graph_algorithms/Flight_Routes_Check.py

def dfs(node, graph, visited):
    visited[node] = True
    for neighbor in graph[node]:
        if not visited[neighbor]:
            dfs(neighbor, graph, visited)

n, m = map(int, input().split())
graph = [[] for _ in range(n + 1)]
for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)

visited = [False] * (n + 1)
dfs(1, graph, visited)

if all(visited[1:]):
    print("YES")
else:
    for city in range(1, n + 1):
        if not visited[city]:
            print("NO")
            print("1", city)
            break
